Pdf4QtLib/sources/pdfparser.cpp

```diff
@@ -163,7 +163,7 @@ PDFLexicalAnalyzer::Token PDFLexicalAnalyzer::fetch()
                 real = -real;
             }
 
-            return !treatAsReal ? Token(TokenType::Integer, integer) : Token(TokenType::Real, real);
+            return !treatAsReal ? Token(TokenType::Integer, QVariant(static_cast<qint64>(integer))) : Token(TokenType::Real, real);
```

Comment:
Strange, why automatic conversion does not worked?

Reply:
PDFInteger is an int64_t which if a typedef long int when __WORDSIZE == 64
The standard also says that the minimum size for a long int is 32 bits.
QVariant doesn't have a constructor with a long int, only for int/uint/qlonglong/qulonglong. Which are all in 64 bits anyway.
I guess gcc does not check the real size of the words.